### PR TITLE
Add commandline option to simulate alert queue expand failure - v2

### PR DESCRIPTION
**Pull request by jufajardini**
Previous PR: #7403 

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/5319

Describe changes:
- add the command-line argument ``--simulate-alert-queue-realloc-failure`` as part of the exception utils options

suricata-verify-pr: 850
https://github.com/OISF/suricata-verify/pull/850

### CODE REVIEW

**Review by victorjulien — APPROVED** [review]

**Comment by jufajardini** [comment]
Force pushed to fix the commit checks, as I had fixed the include typo but in the wrong commit... Sorry about that >__<

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 7791

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 7793

File: src/detect-engine-alert.c [hunk 1]
```diff
@@ -27,6 +27,7 @@
 #include "flow.h"
 #include "flow-private.h"
 
+#include "util-exception-policy.h"
```
[the remark this answers is not in the document]
Comment (jufajardini) [reply src/detect-engine-alert.c:30]: Following Victor's example, this wasn't there, but I think it is a good addition, indeed.

**Comment by jufajardini** [comment]
replaced by: https://github.com/OISF/suricata/pull/7622

**Pull request closed without merging**